src/ProgressOnderwijsUtils.Analyzers/MustUseExpressionResultAnalyzer.cs

```diff
@@ -77,22 +102,39 @@ static bool ExpressionTypeCanBeIgnored(ITypeSymbol? exprType)
                 return true;
             } else if (exprType.SpecialType == SpecialType.System_Void) {
                 return true;
-            } else if (IsUtilsUnit(exprType)) {
+            } else if (IsSymbolOfType(exprType, "ProgressOnderwijsUtils.Collections", "Unit")) {
                 return true;
             } else {
                 // TODO: first start wilt the maybe's, then apply to other/all types
-                return !(exprType.TypeKind == TypeKind.Struct && IsInUtilsCollectionsNamespace(exprType) && exprType.Name == "Maybe");
+                return !IsSymbolOfType(exprType, "ProgressOnderwijsUtils.Collections", "Maybe");
             }
         }
 
-        static bool IsUtilsUnit(ITypeSymbol exprType)
-            => exprType.TypeKind == TypeKind.Struct
-                && IsInUtilsCollectionsNamespace(exprType)
-                && exprType.Name == "Unit";
+        static bool IsSymbolOfType(ITypeSymbol? symbol, string ns, string name)
+        {
+            if (symbol == null) {
+                return false;
+            }
+
+            if (Namespace(symbol) != ns) {
+                return false;
+            }
+
+            return symbol.Name == name;
+        }
 
-        static bool IsInUtilsCollectionsNamespace(ITypeSymbol exprType)
-            => exprType.ContainingNamespace.Name == "Collections"
-                && exprType.ContainingNamespace.ContainingNamespace.Name == "ProgressOnderwijsUtils"
-                && exprType.ContainingNamespace.ContainingNamespace.ContainingNamespace.IsGlobalNamespace;
+        static string Namespace(ITypeSymbol symbol)

```

Comment:
Dit roep je wel aan op potentieel heeel veeel lambda's; het is voorstelbaar dat dit compile-time invloed heeft.

Reply:
(Ik zou af en toe kijken of de build sneller is zonder de analyzer, en zo ja, eens gaan kijken of we hier niet iets te onhandige shortcuts nemen).